{"name": "删除批量创建而产生的部门", "status": "failed", "statusDetails": {"message": "AssertionError: 断言失败 -|- 实际结果:部门下有子部门或成员 || 预期结果: 成功", "trace": "cls = <class 'tools.data_process.DataProcess'>\nresponse = {'errcode': '210008', 'errmsg': '部门下有子部门或成员'}\nexpect_str = '{\"$.errmsg\":\"成功\"}'\n\n    @classmethod\n    def assert_result(cls, response: dict, expect_str: str):\n        \"\"\" 预期结果实际结果断言方法\n        :param response: 实际响应结果\n        :param expect_str: 预期响应内容，从excel中读取\n        return None\n        \"\"\"\n        # 后置sql变量转换\n        expect_str = rep_expr(expect_str, DataProcess.response_dict)\n        expect_dict = convert_json(expect_str)\n        index = 0\n        for k, v in expect_dict.items():\n            # 获取需要断言的实际结果部分\n            actual = extractor(response, k)\n            index += 1\n            logger.info(f'第{index}个断言,实际结果:{actual} | 预期结果:{v} \\n断言结果 {actual == v}')\n            allure_step(f'第{index}个断言',  f'实际结果:{actual} = 预期结果:{v}')\n            try:\n>               assert actual == v\nE               AssertionError\n\ntools/data_process.py:118: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\ncases = ['case_046', '测试数据清理', '成员，部门相关', '删除批量创建而产生的部门', '/v1/dept/delete?access_token=&$.case_001.access_token&', '', ...]\n\n    @pytest.mark.flaky(reruns=0, reruns_delay=1)\n    def test_main(cases):#cases, get_db\n        # 此处的cases入参来自与 conftest.py  文件中 cases函数，与直接使用 @pytest.mark.parametrize\n        # 有着差不多的效果\n        # 发送请求\n        response, expect, sql = BaseRequest.send_request(cases)\n        # 执行sql\n        # DataProcess.handle_sql(sql, get_db)\n        # 断言操作\n>       DataProcess.assert_result(response, expect)\n\ntest/test_api.py:29: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ncls = <class 'tools.data_process.DataProcess'>\nresponse = {'errcode': '210008', 'errmsg': '部门下有子部门或成员'}\nexpect_str = '{\"$.errmsg\":\"成功\"}'\n\n    @classmethod\n    def assert_result(cls, response: dict, expect_str: str):\n        \"\"\" 预期结果实际结果断言方法\n        :param response: 实际响应结果\n        :param expect_str: 预期响应内容，从excel中读取\n        return None\n        \"\"\"\n        # 后置sql变量转换\n        expect_str = rep_expr(expect_str, DataProcess.response_dict)\n        expect_dict = convert_json(expect_str)\n        index = 0\n        for k, v in expect_dict.items():\n            # 获取需要断言的实际结果部分\n            actual = extractor(response, k)\n            index += 1\n            logger.info(f'第{index}个断言,实际结果:{actual} | 预期结果:{v} \\n断言结果 {actual == v}')\n            allure_step(f'第{index}个断言',  f'实际结果:{actual} = 预期结果:{v}')\n            try:\n                assert actual == v\n            except AssertionError:\n>               raise AssertionError(f'断言失败 -|- 实际结果:{actual} || 预期结果: {v}')\nE               AssertionError: 断言失败 -|- 实际结果:部门下有子部门或成员 || 预期结果: 成功\n\ntools/data_process.py:120: AssertionError"}, "steps": [{"name": "请求地址", "status": "passed", "attachments": [{"name": "请求地址", "source": "ddb4e0cf-0b87-4669-87a8-bf17f9a05f20-attachment.txt", "type": "text/plain"}], "start": 1615871743853, "stop": 1615871743853}, {"name": "请求头", "status": "passed", "attachments": [{"name": "请求头", "source": "88dd6e42-f876-4f3e-8a8a-1fb8b64b174e-attachment.txt", "type": "text/plain"}], "start": 1615871743853, "stop": 1615871743854}, {"name": "请求参数", "status": "passed", "attachments": [{"name": "请求参数", "source": "274b45ce-c47a-40fe-b756-9461636969c2-attachment.txt", "type": "text/plain"}], "start": 1615871743854, "stop": 1615871743854}, {"name": "上传文件", "status": "passed", "attachments": [{"name": "上传文件", "source": "d931bbd7-8ef7-4578-8d1c-657f23b0569b-attachment.txt", "type": "text/plain"}], "start": 1615871743854, "stop": 1615871743854}, {"name": "响应耗时(s)", "status": "passed", "attachments": [{"name": "响应耗时(s)", "source": "40449646-647a-4a0a-adeb-a70969b02f73-attachment.txt", "type": "text/plain"}], "start": 1615871744090, "stop": 1615871744090}, {"name": "响应内容", "status": "passed", "attachments": [{"name": "响应内容", "source": "e9d5e86f-4453-478d-ae5a-351d998579dc-attachment.txt", "type": "text/plain"}], "start": 1615871744090, "stop": 1615871744091}, {"name": "存储实际响应", "status": "passed", "attachments": [{"name": "存储实际响应", "source": "3c4c7b2d-966e-4079-a50e-7c66b4603837-attachment.txt", "type": "text/plain"}], "start": 1615871744091, "stop": 1615871744092}, {"name": "第1个断言", "status": "passed", "attachments": [{"name": "第1个断言", "source": "1421072d-5385-4f6b-bd0d-5f21523b8721-attachment.txt", "type": "text/plain"}], "start": 1615871744093, "stop": 1615871744093}], "parameters": [{"name": "cases", "value": "['case_046', '测试数据清理', '成员，部门相关', '删除批量创建而产生的部门', '/v1/dept/delete?access_token=&$.case_001.access_token&', '', 'get', 'params', '', '{\\n \"did\": \"&$.case_045.datas[1].did&\"\\n}', '', '', '{\"$.errmsg\":\"成功\"}']"}], "start": 1615871743853, "stop": 1615871744093, "uuid": "205c2836-bc46-4e86-b263-53021ce9c859", "historyId": "7bd6a1d0f5bf41f56aee1bfbd64df136", "testCaseId": "881516e7ef2e2aa5436f6a3c3e6a5bcb", "fullName": "test.test_api#test_main", "labels": [{"name": "tag", "value": "flaky(reruns=0, reruns_delay=1)"}, {"name": "parentSuite", "value": "test"}, {"name": "suite", "value": "test_api"}, {"name": "host", "value": "liuchengxudeMacBook-Pro.local"}, {"name": "thread", "value": "3134-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test.test_api"}]}